x-pack/plugins/security_solution/cypress/integration/detection_rules/bulk_edit_rules.spec.ts

```diff
@@ -90,180 +130,358 @@ describe('Detection rules, bulk edit', () => {
+  describe('Index patterns', () => {
+    it('Index pattern action applied to custom rules, including machine learning: user proceeds with edit of custom non machine learning rule', () => {
+      const indexPattersToBeAdded = ['index-to-add-1-*', 'index-to-add-2-*'];
+      const resultingIndexPatterns = [...prePopulatedIndexPatterns, ...indexPattersToBeAdded];
+
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+      clickAddIndexPatternsMenuItem();
+
+      // confirm editing custom rules, that are not Machine Learning
+      checkMachineLearningRulesCannotBeModified(expectedNumberOfMachineLearningRulesToBeEdited);
+      cy.get(MODAL_CONFIRMATION_BTN).click();
+
+      typeIndexPatterns(indexPattersToBeAdded);
+      submitBulkEditForm();
+
+      waitForBulkEditActionToFinish({ rulesCount: expectedNumberOfNotMLRules });
+
+      // check if rule has been updated
+      goToTheRuleDetailsOf(RULE_NAME);
+      hasIndexPatterns(resultingIndexPatterns.join(''));
+    });
+
+    it('Index pattern action applied to custom rules, including machine learning: user cancels action', () => {
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+      clickAddIndexPatternsMenuItem();
+
+      // confirm editing custom rules, that are not Machine Learning
+      checkMachineLearningRulesCannotBeModified(expectedNumberOfMachineLearningRulesToBeEdited);
+
+      // user cancels action and modal disappears
+      cancelConfirmationModal();
+    });
+
+    it('Add index patterns to custom rules', () => {
+      const indexPattersToBeAdded = ['index-to-add-1-*', 'index-to-add-2-*'];
+      const resultingIndexPatterns = [...prePopulatedIndexPatterns, ...indexPattersToBeAdded];
+
+      // select only rules that are not ML
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+      unselectRuleByName(getMachineLearningRule().name);
+
+      openBulkEditAddIndexPatternsForm();
+      typeIndexPatterns(indexPattersToBeAdded);
+      submitBulkEditForm();
+
+      waitForBulkEditActionToFinish({ rulesCount: expectedNumberOfNotMLRules });
+
+      // check if rule has been updated
+      goToTheRuleDetailsOf(RULE_NAME);
+      hasIndexPatterns(resultingIndexPatterns.join(''));
+    });
+
+    it('Overwrite index patterns in custom rules', () => {
+      const indexPattersToWrite = ['index-to-write-1-*', 'index-to-write-2-*'];
+
+      // select only rules that are not ML
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+      unselectRuleByName(getMachineLearningRule().name);
+
+      openBulkEditAddIndexPatternsForm();
+
+      // check overwrite index patterns checkbox, ensure warning message is displayed and type index patterns
+      checkOverwriteIndexPatternsCheckbox();
+      cy.get(RULES_BULK_EDIT_INDEX_PATTERNS_WARNING).should(
+        'have.text',
+        `You’re about to overwrite index patterns for ${expectedNumberOfNotMLRules} selected rules, press Save to apply changes.`
```

Comment:
This Cypress assertion looks brittle because it depends so exactly on the Ui copy text. I think we could do:
```js
      cy.get(RULES_BULK_EDIT_INDEX_PATTERNS_WARNING).should(
        'contain.text',
        `You’re about to overwrite index patterns for ${expectedNumberOfNotMLRules} selected rules, press Save to apply changes.`
);
```
which would still check the warning to refer to `expectedNumberOfNotMLRules` without relying so explicitly on the text. (Notice the `contain.text` instead of `have.text`)

This applies to similar cases in this test file.

Reply:
I prefer full text assertion, we can make sure the message has exactly the same role, we intend to have.

For example, the goal of this warning message to bring to user's attention that some data can be lost, rather the number of rule that will be affected. That specific number has been asserted before and also after test completed, so we are sure all the rules have been successfully updated.

Of course, it's a tradeoff that's can have it's drawback. I.e. if warning changes, we will need to make changes here as well.